katdal/chunkstore_s3.py

```diff
@@ -0,0 +1,95 @@
+################################################################################
+# Copyright (c) 2011-2016, National Research Foundation (Square Kilometre Array)
+#
+# Licensed under the BSD 3-Clause License (the "License"); you may not use
+# this file except in compliance with the License. You may obtain a copy
+# of the License at
+#
+#   https://opensource.org/licenses/BSD-3-Clause
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+"""A store of chunks (i.e. N-dimensional arrays) based on the Amazon S3 API."""
+
+import io
+
+import numpy as np
+try:
+    import botocore
+except ImportError:
+    botocore = None
+else:
+    import botocore.config
+    import botocore.session
```

Comment:
I'll think about it. It depends on how the user accesses these classes (via top level or not).

Reply:
Fair enough. I think I ran into the same problem with katsdpservices and didn't find a nice solution. It might be a good idea to save away the `ImportError` object and reraise it when the class is instantiated, so that the user gets the full details of what was missing (e.g. it might be failure to import some dependency of boto/rados rather than the package itself).